demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1974 @@
+
+    xCommandParams.blockTimeMs = MQTT_AGENT_SEND_BLOCK_TIME_MS;
+    xCommandParams.cmdCompleteCallback = prvMQTTAgentCmdCompleteCallback;
+    xCommandParams.pCmdCompleteCallbackContext = &xCommandContext;
+    xCommandContext.xTaskToNotify = xTaskGetCurrentTaskHandle();
+    xCommandContext.pArgs = NULL;
+    xCommandContext.xReturnStatus = MQTTSendFailed;
+
+    xCommandStatus = MQTTAgent_Publish( &xGlobalMqttAgentContext, &xPublishInfo, &xCommandParams );
+    configASSERT( xCommandStatus == MQTTSuccess );
+
+    xTaskNotifyWait( 0,
+                     0,
+                     NULL,
+                     pdMS_TO_TICKS( MQTT_AGENT_MS_TO_WAIT_FOR_NOTIFICATION ) );
+
+    if( xCommandContext.xReturnStatus != MQTTSuccess )
+    {
+        LogError( ( "Failed to send PUBLISH packet to broker with error = %u.", xCommandContext.xReturnStatus ) );
+        xOtaMqttStatus = OtaMqttPublishFailed;
+    }
+    else
+    {
+        LogInfo( ( "Sent PUBLISH packet to broker %.*s to broker.\n\n",
+                   usTopicLen,
+                   pcTopic ) );
+    }
+
+    return xOtaMqttStatus;
+}
+/*-----------------------------------------------------------*/
+
+static OtaMqttStatus_t prvMqttUnSubscribe( const char * pcTopicFilter,
+                                           uint16_t usTopicFilterLength,
+                                           uint8_t ucQOS )
+{
+    OtaMqttStatus_t xOtaMqttStatus = OtaMqttSuccess;
+    MQTTStatus_t xCommandStatus;
+    MQTTAgentCommandInfo_t xCommandParams = { 0 };
+    MQTTAgentCommandContext_t xCommandContext = { 0 };
+    MQTTSubscribeInfo_t xSubscription = { 0 };
+    MQTTAgentSubscribeArgs_t xSubscribeArgs = { 0 };
+
+    xSubscription.qos = ucQOS;
+    xSubscription.pTopicFilter = pcTopicFilter;
+    xSubscription.topicFilterLength = usTopicFilterLength;
+    xSubscribeArgs.numSubscriptions = 1;
+    xSubscribeArgs.pSubscribeInfo = &xSubscription;
+
+
+    xCommandParams.blockTimeMs = MQTT_AGENT_SEND_BLOCK_TIME_MS;
+    xCommandParams.cmdCompleteCallback = prvMQTTUnsubscribeCompleteCallback;
+    xCommandParams.pCmdCompleteCallbackContext = &xCommandContext;
+    xCommandContext.xTaskToNotify = xTaskGetCurrentTaskHandle();
+    xCommandContext.pArgs = &xSubscribeArgs;
+    xCommandContext.xReturnStatus = MQTTSendFailed;
+
+    /* Unsubscribe to topics. */
+    xCommandStatus = MQTTAgent_Unsubscribe( &xGlobalMqttAgentContext, &xSubscribeArgs, &xCommandParams );
+    configASSERT( xCommandStatus == MQTTSuccess );
+
+    xTaskNotifyWait( 0,
+                     0,
+                     NULL,
+                     pdMS_TO_TICKS( MQTT_AGENT_MS_TO_WAIT_FOR_NOTIFICATION ) );
+
+    if( xCommandContext.xReturnStatus != MQTTSuccess )
+    {
+        LogError( ( "Failed to send unsubsribe packet to broker with error = %u.", xCommandContext.xReturnStatus ) );
+        xOtaMqttStatus = OtaMqttSubscribeFailed;
+    }
+    else
+    {
+        LogInfo( ( "Unsubsribed from topic %.*s to broker.\n\n",
```

Comment:
```suggestion
        LogInfo( ( "Unsubscribed from topic %.*s to broker.\n\n",
```

Reply:
Updated in https://github.com/aws/amazon-freertos/pull/3188/commits/450fda5aa67e04ef89b710317773948af5a9259c